lib/interactor/error.rb

```diff
@@ -28,4 +28,39 @@ def initialize(context = nil)
       super
     end
   end
+
+  class ContractError < StandardError
+    attr_reader :context
+
+    def initialize(context=nil, opts={})
+      @context = context
+      @message = opts[:message]
+      @undeclared_properties = opts[:undeclared] || []
+      @missing_properties = opts[:missing] || []
+      super()
```

Comment:
I do actually need the parens. Otherwise, it calls `super` with two extra args (i.e. `context` and `opts`).